### PR TITLE
[GLUTEN-1379] [CH] update clickhouse backends doc

### PR DESCRIPTION
**Pull request by liuneng1994**
## What changes were proposed in this pull request?

update clickhouse backends doc

Fixes: #1379 

## How was this patch tested?

Only doc.

### CODE REVIEW

File: docs/ClickHouse.md [hunk 1]
```diff
@@ -68,13 +68,27 @@ In general, we use IDEA for Gluten development and CLion for ClickHouse backend
 - Open Gluten code in IDEA
 
 #### Setup ClickHouse backend development environment
+Support the use of existing ClickHouse projects for development, and also support automatic clone ClickHouse to cpp-ch.
 
+The following is an example of using an external warehouse to configure the development environment. Non-existing CH sources can skip the clone stage of ClickHouse.
```
Comment (baibaichen) [inline docs/ClickHouse.md:73]: warehouse  => repo?

Comment (baibaichen) [inline docs/ClickHouse.md:73]: > Non-existing CH sources can skip the clone stage of ClickHouse.

what is  does **Non-existing CH sources** mean?

File: docs/ClickHouse.md [hunk 2]
```diff
@@ -98,8 +112,18 @@ In general, we use IDEA for Gluten development and CLion for ClickHouse backend
 ### Compile ClickHouse backend
 First need to enter the root directory of the Gluten project.
 run`sudo ./ep/build-clickhouse/src/install_ubuntu.sh`,Install the software required for compilation.  
-Create a build directory, such as /tmp/build_clickhouse, run `./ep/build-clickhouse/src/build_clickhouse.sh --src = /path /to/clickhouse --build_dir=/tmp/build_clickhouse`.  
-Target file is `/tmp/build_clickhouse/utils/local-engine/libch.so`.   
+Create a build directory, such as /tmp/build_clickhouse, run `./ep/build-clickhouse/src/build_clickhouse.sh --src = /path/to/gluten`.
```
Comment (baibaichen) [inline docs/ClickHouse.md:115]: How `/tmp/build_clickhouse` is used?

Comment (liuneng1994) [reply docs/ClickHouse.md]: remove

**Review by baibaichen — APPROVED** [review]

**Pull request merged**